pkg/streamingpromql/optimize/ast/prune_toggles.go

```diff
@@ -16,13 +16,29 @@ import (
 // e.g. `(avg(rate(foo[1m]))) and on() (vector(1) == 1)` -> `(avg(rate(foo[1m])))`
 // e.g. `(avg(rate(foo[1m]))) and on() (vector(1) == -1)` -> `(vector(1) == -1)`
 type PruneToggles struct {
-	mapper astmapper.ASTMapper
+	mapper *ASTExprMapperWithState
+
+	pruneTogglesMetrics
```

Comment:
Is there a reason why this is embedded rather than an ordinary field?

Reply:
No particular reason, but I saw other files with this pattern and liked the way it was organised (metrics separate from the rest), but am open to changing it.

Reply:
I'd err on the side of making it an ordinary field unless there's a compelling reason to make it embedded.